File: src/FrameAuth/Models/UserViewModels/UserViewModel.cs
using System;

namespace FrameAuth.Models.UserViewModels
{
    public enum LanguageIso
    {
        en = 0,
        hu,
        de
    }

    public interface IUserViewModel
    {
        int AccessFailedCount { get; set; }
        string Email { get; set; }
        bool EmailConfirmed { get; set; }
        Guid Id { get; set; }
        bool LockoutEnabled { get; set; }
        DateTimeOffset? LockoutEnd { get; set; }
        string NormalizedEmail { get; set; }
        string NormalizedUserName { get; set; }
        string PhoneNumber { get; set; }
        bool PhoneNumberConfirmed { get; set; }
        bool TwoFactorEnabled { get; set; }
        string UserName { get; set; }
        bool isAdmin { get; set; }
        LanguageIso Language { get; set; }
    }

    public class UserViewModel: IUserViewModel
    {
        public int AccessFailedCount { get; set; }
        public string Email { get; set; }
        public bool EmailConfirmed { get; set; }
        public Guid Id { get; set; }
        public bool LockoutEnabled { get; set; }
        public DateTimeOffset? LockoutEnd { get; set; }
        public string NormalizedEmail { get; set; }
        public string NormalizedUserName { get; set; }
        public string PhoneNumber { get; set; }
        public bool PhoneNumberConfirmed { get; set; }
        public bool TwoFactorEnabled { get; set; }
        public string UserName { get; set; }
        public bool isAdmin { get; set; }
        public LanguageIso Language { get; set; }
    }
}

File: src/FrameUI/src/app/models/UserViewModel.ts

export enum LanguageIso {
    en = 0,
    hu = 1,
    de = 2    
}

export interface IuserViewModel {
    accessFailedCount: number;
    email: string;
    emailConfirmed: boolean;
    id: string;
    lockoutEnabled: boolean;
    lockoutEnd: Date;
    normalizedEmail: string;
    normalizedUserName: string;
    phoneNumber: string;
    phoneNumberConfirmed: boolean;
    twoFactorEnabled: boolean;
    userName: string;
    isAdmin: boolean;
    language: LanguageIso;
    
}

export class UserViewModel implements IuserViewModel {
    public accessFailedCount: number;
    
    public email: string;
    
    public emailConfirmed: boolean;
    
    public id: string;
    
    public lockoutEnabled: boolean;
    
    public lockoutEnd: Date;
    
    public normalizedEmail: string;
    
    public normalizedUserName: string;
    
    public phoneNumber: string;
    
    public phoneNumberConfirmed: boolean;
    
    public twoFactorEnabled: boolean;
    
    public userName: string;
    
    public isAdmin: boolean;
    
    public language: LanguageIso;
    
    constructor(model?: IuserViewModel) {
        if(model) {
            this.accessFailedCount = model.accessFailedCount;
            this.email = model.email;
            this.emailConfirmed = model.emailConfirmed;
            this.id = model.id;
            this.lockoutEnabled = model.lockoutEnabled;
            this.lockoutEnd = model.lockoutEnd;
            this.normalizedEmail = model.normalizedEmail;
            this.normalizedUserName = model.normalizedUserName;
            this.phoneNumber = model.phoneNumber;
            this.phoneNumberConfirmed = model.phoneNumberConfirmed;
            this.twoFactorEnabled = model.twoFactorEnabled;
            this.userName = model.userName;
            this.isAdmin = model.isAdmin;
            this.language = model.language;
            
        }
    }
}
 